### PR TITLE
Feature/proctree query time

### LINKED ISSUE #3700: Process Tree queries are missing query time
The queries of the Process Tree are giving information according to a query time.
This means that the query time is very relevant in the context of the information.
For example, given a specific process information, you might want to query the matching task information (aka thread information). However, to get the info relevant to the given process info, you must query the exact same time. This is very hard to do in a standalone functions without the timestamp integration into the information given. If this info will move between different code components, we can't expect the use to couple them together himself, as it is part of the info.
This problem is even more severe with the lineage query, which gives each ancestor in a different time (according to fork time of its child).
I think that some sort of coupling of the information with the query time is required.

### PR DESCRIPTION
**Pull request by AlonZivony**
### 1. Explain what the PR does

7fa990fb7 **feat(proctree): fill timestamp of proctree info objects**

Fix #3700

### 2. Explain how to test it

e2e test

### CODE REVIEW

**Comment by AlonZivony** [comment]
@rafaeldtinoco WDYT?

File: go.mod [hunk 1]
```diff
@@ -181,4 +181,5 @@ require (
 	kernel.org/pub/linux/libs/security/libcap/psx v1.2.68 // indirect
 )
 
+replace github.com/aquasecurity/tracee/types => ./types
```
Comment (rafaeldtinoco) [inline go.mod:184]: temporary, needs a PR type change (just marking here not to forget).

File: tests/e2e-inst-signatures/e2e-proctree_data_source.go [hunk 2]
```diff
@@ -161,6 +162,9 @@ func (sig *e2eProcessTreeDataSource) checkThread(eventObj *trace.Event) error {
 	if threadInfo.Pid != eventObj.HostProcessID {
 		return fmt.Errorf(debug("no match for pid"))
 	}
+	if threadInfo.Timestamp != queryTime {
+		return fmt.Errorf(debug("no match for info timestamp"))
+	}
```
Comment (rafaeldtinoco) [inline tests/e2e-inst-signatures/e2e-proctree_data_source.go:167]: Code is correct. Logic is correct, etc. But, why ? The issue does not describe the "why" very well.

If you query for a process, or a thread, or a tree, at a specific point in time, you will get the nearest changelog timestamp item. Why there is a need to know the exact point in time ? (Im not saying I'm against). What would be the benefit ?

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
LGTM. Needs a better reasoning and a PR changing the types before this can be merged. Im putting a -1 for now until both are satisfied. Request my review in the types change, please. I'll merge and you can rebase this and change types dependency to that commit.

**Comment by AlonZivony** [comment]
@rafaeldtinoco I probably need your opinion on the following question:
Should the query time be part of the info objects, or be part of an enveloping type to the info objects?
In my opinion it is easier to manage as part of the info object, but I am not sure that the timestamp is an info of the process in a way that justify this coupling.
We can create a generic `TimeRelevantInfo` type, that has information and timestamp in it, and return it instead of the process and thread info objects.
WDYT?

**Comment by rafaeldtinoco** [comment]
> We can create a generic `TimeRelevantInfo` type, that has information and timestamp in it, and return it instead of the process and thread info objects.

I think I prefer this idea better Alon, if you also think that.

**Review by rafaeldtinoco — CHANGES_REQUESTED** [review]
We agreed the location of Timestamp will be a struct called TimeRelevantInfo instead of duplicates under other Info structs.

**Review by rafaeldtinoco — APPROVED** [review]
LGTM

**Pull request merged**